src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -3,202 +3,422 @@
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                    x,
+                };
+
+                return Unsafe.AsRef<Vector128<byte>>(pResult);
+            }
         }
 
         /// <summary>Creates a new <see cref="Vector128{Double}" /> instance with all elements initialized to the specified value.</summary>
         /// <param name="value">The value that all elements will be initialized to.</param>
         /// <returns>A new <see cref="Vector128{Double}" /> with all elements initialized to <paramref name="value" />.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector128<double> Create(double value)
         {
-            var pResult = stackalloc double[2]
+            if (Sse3.IsSupported)
             {
-                value,
-                value,
-            };
+                Vector128<double> result = CreateScalarUnsafe(value);                       // < v, ? >
+                return Sse3.MoveAndDuplicate(result);                                       // < v, v >
+            }
 
-            return Unsafe.AsRef<Vector128<double>>(pResult);
+            if (Sse.IsSupported)
+            {
+                Vector128<double> result = CreateScalarUnsafe(value);                       // < v, ? >
+                return Sse.MoveLowToHigh(result.AsSingle(), result.AsSingle()).AsDouble();  // < v, v >
+            }
+
+            return CreateSoftware(value);
+
+            Vector128<double> CreateSoftware(double x)
+            {
+                var pResult = stackalloc double[2]
+                {
+                    x,
+                    x,
+                };
+
+                return Unsafe.AsRef<Vector128<double>>(pResult);
+            }
         }
 
         /// <summary>Creates a new <see cref="Vector128{Int16}" /> instance with all elements initialized to the specified value.</summary>
         /// <param name="value">The value that all elements will be initialized to.</param>
         /// <returns>A new <see cref="Vector128{Int16}" /> with all elements initialized to <paramref name="value" />.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector128<short> Create(short value)
         {
-            var pResult = stackalloc short[8]
+            if (Avx2.IsSupported)
             {
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-                value,
-            };
+                Vector128<short> result = CreateScalarUnsafe(value);                        // < v, ?, ?, ?, ?, ?, ?, ? >
```

Comment:
It seems like a strange limitation that `movsx`/`movzx` would be "required" when not inlined but could be elided when it is inlined.

Reply:
> It seems like a strange limitation that movsx/movzx would be "required" when not inlined but could be elided when it is inlined.

This has to do with the fact that parameters are `normalizeOnLoad`, because we can't assume that it is appropriately extended. Once it's inlined, the parameters become locals, which are `normalizeOnStore`.